polars-plan does not compile with concat_str and lazy features

### Checks

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```bash
cargo new polars-compile-error && cd polars-compile-error
cargo add polars -F lazy,concat_str
cargo check
```

### Log output

```shell
error[E0599]: no variant or associated item named `StringExpr` found for enum `dsl::function_expr::FunctionExpr` in the current scope
   --> crates/polars-plan/src/plans/conversion/dsl_to_ir/expr_expansion.rs:117:16
    |
117 |             F::StringExpr(StringFunction::ConcatHorizontal { .. })
    |                ^^^^^^^^^^ variant or associated item not found in `dsl::function_expr::FunctionExpr`
    |
   ::: crates/polars-plan/src/dsl/function_expr/mod.rs:75:1
    |
 75 | pub enum FunctionExpr {
    | --------------------- variant or associated item `StringExpr` not found for this enum
    |
help: there is a variant with a similar name
    |
117 -             F::StringExpr(StringFunction::ConcatHorizontal { .. })
117 +             F::StructExpr(StringFunction::ConcatHorizontal { .. })
    |

error[E0433]: failed to resolve: use of undeclared type `StringFunction`
   --> crates/polars-plan/src/plans/conversion/dsl_to_ir/expr_expansion.rs:117:27
    |
117 |             F::StringExpr(StringFunction::ConcatHorizontal { .. })
    |                           ^^^^^^^^^^^^^^
    |                           |
    |                           use of undeclared type `StringFunction`
    |                           help: an enum with a similar name exists: `StructFunction`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `polars-plan` (lib) due to 2 previous errors
```

### Issue description

Discovered using `cargo-hack`.

Compilation fails when using `lazy` and `concat_str` features without also using the `strings` feature.

### Root Cause

In `crates/polars-plan/src/plans/conversion/dsl_to_ir/expr_expansion.rs`, the code attempts to match against `StringFunction::ConcatHorizontal` with only the `concat_str` feature guard:

```rust
#[cfg(feature = "concat_str")]
{
    use crate::dsl::StringFunction;
    expand_into_inputs |= matches!(
        function,
        F::StringExpr(StringFunction::ConcatHorizontal { .. })
    );
}
```


### Expected behavior

Crate compiles successfully.

### Installed versions

- OS: Rocky Linux 8.6
- rustc: 1.92.0-nightly (6501e64fc 2025-10-23)
- polars: 0.52.0 or current main: fea5335e5b
- features: concat_str,lazy

--- a/crates/polars-plan/src/plans/conversion/dsl_to_ir/expr_expansion.rs
+++ b/crates/polars-plan/src/plans/conversion/dsl_to_ir/expr_expansion.rs
@@ -110,8 +110,9 @@ fn function_input_wildcard_expansion(function: &FunctionExpr) -> FunctionExpansi
         expand_into_inputs |= matches!(function, F::FfiPlugin { flags, .. } if flags.flags.contains(FunctionFlags::INPUT_WILDCARD_EXPANSION));
         allow_empty_inputs |= matches!(function, F::FfiPlugin { flags, .. } if flags.flags.contains(FunctionFlags::ALLOW_EMPTY_INPUTS));
     }
-    #[cfg(feature = "concat_str")]
+    #[cfg(all(feature = "strings", feature = "concat_str"))]
     {
+        use crate::dsl::StringFunction;
         expand_into_inputs |= matches!(
             function,
             F::StringExpr(StringFunction::ConcatHorizontal { .. })

